lib/collection/widgets/second_pass_food_item_card.dart

```diff
@@ -19,101 +20,103 @@ class SecondPassFoodItemCard extends StatelessWidget {
-          child: Text('Soaking and stir-fried'),
-          value: 'Soaking and stir-fried'),
-      DropdownMenuItem(
-          child: Text('Boiled and fried'), value: 'Boiled and fried'),
-      DropdownMenuItem(
-          child: Text('Boiled and stir-fried'), value: 'Boiled and stir-fried'),
-      DropdownMenuItem(
-          child: Text('Steamed and fried'), value: 'Steamed and fried'),
-      DropdownMenuItem(
-          child: Text('Roasted and boiled'), value: 'Roasted and boiled'),
-      DropdownMenuItem(child: Text('Other'), value: 'Other')
+    const List<String> preparationMethods = [
+      'Raw',
+      'Boiled',
+      'Boiled in water but retained water',
+      'Boiled in water but removed water',
+      'Steamed',
+      'Roasted with oil',
+      'Roasted without oil',
+      'Fried',
+      'Stir-fried',
+      'Soaking and stir-fried',
+      'Boiled and fried',
+      'Boiled and stir-fried',
+      'Steamed and fried',
+      'Roasted and boiled',
+      'Other'
     ];
 
     return Card(
       child: Padding(
         padding: const EdgeInsets.all(8.0),
         child: Column(
           children: [
-            Text(foodItem.name.value +
+            Text(foodItemDisplayName +
                 ' consumed in the ' +
-                foodItem.timePeriod.value),
-            // TODO: fix RenderFlex overflow
-            DropdownButtonFormField(
-              value: foodItem.source.value,
-              decoration: InputDecoration(
-                icon: const Icon(Icons.kitchen),
-                labelText: 'Food source',
-                helperText: 'Where does the food come from',
-                errorText:
-                    foodItem.source.invalid ? 'Select the food source' : null,
-              ),
-              items: sourceDropdownMenuItems,
-              onChanged: (String? value) => onSourceChanged!(value ?? ''),
-            ),
+                foodItemDisplayTimePeriod),
+            DropdownSearch<String>(
+                maxHeight: 448.0,
+                dropdownSearchDecoration: InputDecoration(
+                    icon: const Icon(Icons.kitchen),
+                    labelText: 'Food source',
+                    helperText: 'Where does the food come from',
+                    // TODO: the errorText should be displayed if nothing is chosen
+                    // so investigate how this can be achieved with focusnodes or
+                    // maybe send an empty string (although that would not work all
+                    // the time), currently errorText is never displayed
+                    errorText: isFieldModifiedAndEmpty(foodItem.source)
+                        ? 'Select the food source'
+                        : null),
+                mode: Mode.MENU,
+                showSelectedItems: true,
+                showSearchBox: false,
+                items: foodSources,
+                onChanged: (String? foodSource) =>
+                    onSourceChanged!(foodSource ?? ''),
```

Comment:
Not sure which lines from the highlighted 81-84. I'm assuming you mean lines 83 and 84 — the `foodSource ?? ''` is needed as `onSourceChanged` only accepts non-nullable Strings as arguments. But lmk if this is what you meant, I'm not quite sure.

Reply:
More meant something like below since foodSource would not be null given the selection options
```suggestion
                    onSourceChanged!(foodSource!),
```

Reply:
Done! Thanks 😊